hw/ip/otbn/dv/model/iss_wrapper.cc

```diff
@@ -322,9 +334,17 @@ void ISSWrapper::start(uint32_t addr) {
   run_command(oss.str(), nullptr);
 }
 
-std::pair<bool, uint32_t> ISSWrapper::step(bool gen_trace) {
+std::pair<bool, uint32_t> ISSWrapper::step(bool edn_rnd_data_valid,
+                                           uint32_t edn_rnd_data[8],
+                                           bool gen_trace) {
+  std::ostringstream oss;
+  oss << "step";
+  if (edn_rnd_data_valid) {
+    oss << " edn_rnd_data=" << wlen_val_to_hex_str(edn_rnd_data);
+  }
```

Comment:
I think this is probably more complicated than it needs to be. Can you do it as two ISS commands instead? Something like `edn_data 0x12345`, followed by `step`? I think this will make lots of the rest of the Python changes much simpler.

Reply:
I can do, it'll simplify the handling of the `step` command (removing argument parsing) though there is a concern with timing. When `edn_data` gets called will have an effect on how the stalling behaviour works in the ISS. We can of course ensure we're doing that properly but there will be no way for the ISS to report `edn_data` has been called at the wrong point. The current everything in `step` we just have a single command that does everything required in a cycle and can ensure timing of the RND update vs the instruction execution/stall occurs correctly.

Either way I think we still have the issue of how we want to deal with stalling RND reads within the ISS.

Reply:
Well, I was imagining that the ISS state would have some sort of "pending" flag for stuff that just came in from the EDN, so all the complicated timing/stalling logic can read that. I don't think we to thread all the inputs through the various method calls like this patch currently does.

Reply:
Oh, (and this is obvious, but...) please can you make sure that you add any non-obvious timing details to the spec as well as the ISS? The idea is that a reader could re-create the ISS from reading the spec: we really don't want to end up modelling the RTL in the golden reference without at least documenting what we've done.

Reply:
We can do things that way, but the crux of the issue is `step` is effectively telling the ISS it's time to process a clock cycle so it's crucial that `step` and `edn_rnd_data` is called in a consistent order. E.g. For any given clock cycle either `edn_rnd_data` should be called before `step`. Depending on when you call it will alter the timing of which `step` sees the pending flag being set.

My slight worry is should the order somehow get flipped around it'll break things but there's no check that can tell us we've messed it up. Still I guess we don't call `step` from many places so just need to add a comment warning the order is important. We're probably best off having the `otbn_model_step` DPI call still take in the `edn_rnd_data` and `edn_rnd_data_valid` signals and then separating into two calls inside `otbn_model.cc`. Otherwise we've got a situation where the order of a function call vs a non-blocking assignment in `otbn_core_model.sv` is important.

I'll add some information about how RND stalls relate to the EDN interface to the documentation.

Reply:
Er, don't we just assume that `step` is the thing that consumes simulation time and that any other thing that should apply to that clock cycle needs to come first? I don't really see a problem here.

Reply:
Yes, that is what I am going to do.